src/Symfony/Component/HttpKernel/Event/ControllerArgumentsEvent.php

```diff
@@ -23,6 +26,27 @@
  *
  * @author Christophe Coevoet <stof@notk.org>
  */
-class ControllerArgumentsEvent extends FilterControllerArgumentsEvent
+class ControllerArgumentsEvent extends ControllerEvent
 {
+    private $arguments;
+
+    public function __construct(HttpKernelInterface $kernel, callable $controller, array $arguments, Request $request, ?int $requestType)
+    {
+        parent::__construct($kernel, $controller, $request, $requestType);
+
+        $this->arguments = $arguments;
+    }
+
+    /**
+     * @return array
```

Comment:
let's use a real type instead? could require making either the class or the method final in 4.4

Reply:
oh wait sorry I forgot about #30323 which will allow this everywhere!